src/Okta.Sdk.IntegrationTests/UserScenarios.cs

```diff
@@ -763,37 +764,40 @@ public async Task RemoveGroupTargetFromRole()
         [Fact]
         public async Task ForgotPasswordGenerateOneTimeToken()
         {
-            var client = TestClient.Create();
-            var guid = Guid.NewGuid();
-
-            var createdUser = await client.Users.CreateUserAsync(new CreateUserWithPasswordOptions
+            await RunTestWithRetry(nameof(ForgotPasswordGenerateOneTimeToken), async () =>
             {
-                Profile = new UserProfile
+                var client = TestClient.Create();
+                var guid = Guid.NewGuid();
+
+                var createdUser = await client.Users.CreateUserAsync(new CreateUserWithPasswordOptions
                 {
-                    FirstName = "John",
-                    LastName = $"{nameof(ForgotPasswordGenerateOneTimeToken)}",
-                    Email = $"{nameof(ForgotPasswordGenerateOneTimeToken)}-dotnet-sdk-{guid}@example.com",
-                    Login = $"{nameof(ForgotPasswordGenerateOneTimeToken)}-dotnet-sdk-{guid}@example.com",
-                },
-                RecoveryQuestion = "Answer to life, the universe, & everything",
-                RecoveryAnswer = "42 of course",
-                Password = "Abcd1234",
-                Activate = true,
-            });
+                    Profile = new UserProfile
+                    {
+                        FirstName = "John",
+                        LastName = $"{nameof(ForgotPasswordGenerateOneTimeToken)}",
+                        Email = $"{nameof(ForgotPasswordGenerateOneTimeToken)}-dotnet-sdk-{guid}@example.com",
+                        Login = $"{nameof(ForgotPasswordGenerateOneTimeToken)}-dotnet-sdk-{guid}@example.com",
+                    },
+                    RecoveryQuestion = "Answer to life, the universe, & everything",
+                    RecoveryAnswer = "42 of course",
+                    Password = "Abcd1234",
+                    Activate = true,
+                });
 
-            Thread.Sleep(5000); // allow for user replication prior to read attempt
+                Thread.Sleep(5000); // allow for user replication prior to read attempt
 
-            try
-            {
-                var forgotPasswordResponse = await createdUser.ForgotPasswordGenerateOneTimeTokenAsync(false);
-                forgotPasswordResponse.Should().NotBeNull();
-                forgotPasswordResponse.ResetPasswordUrl.Should().NotBeNullOrEmpty();
-            }
-            finally
-            {
-                await createdUser.DeactivateAsync();
-                await createdUser.DeactivateOrDeleteAsync();
-            }
+                try
+                {
+                    var forgotPasswordResponse = await createdUser.ForgotPasswordGenerateOneTimeTokenAsync(false);
```

Comment:
+1 for the retry approach, but I think we should only retry 
`var forgotPasswordResponse = await createdUser.ForgotPasswordGenerateOneTimeTokenAsync(false);` plus asserts since that piece of code seems to be the problematic one. User creation shouldn't be retried.

We have done something similar in the past, so FYI you can use `Polly` that is already installed and does the retry work for you: https://github.com/okta/okta-sdk-dotnet/blob/cb0b36c4732ef75967a3c2d1bca1ea9c766a5ead/src/Okta.Sdk.IntegrationTests/UserScenarios.cs#L60